File: Frontend/src/models/product.ts

export interface Product {
  id: number
  title: string
  description: string
  price: number
  image: string
  category: string
  author: string
  quantityInStock: number
}

export interface ProductParams {
  orderBy: string;
  searchTerm?: string;
  categories: string[];
  authors: string[];
  pageNumber: number;
  pageSize: number;
}

File: Backend/Entities/Product.cs
namespace Backend.Entities;

public class Product
{
    public int Id { get; set; }
    public string Title { get; set; }
    public string Description { get; set; }
    public long Price { get; set; }
    public string Image { get; set; }
    public string Category  { get; set; }
    public string Author { get; set; }
    public int QuantityInStock { get; set; }
    public string PublicId { get; set; }
}